wurst/data structures/LinkedHashMap.wurst

```diff
@@ -0,0 +1,225 @@
+package LinkedHashMap
+import HashMap
+import LinkedList
+import Wurstunit
+
+public class LinkedHashMap<K, V> extends HashMap<K, V>
+    private LinkedList<K>  keys
+    private LHMEntry<K, V> dummy
+
+    construct()
+        keys  = new LinkedList<K>()
+        dummy = new LHMEntry<K, V>(null, null)
+
+
+    /** Saves the given value under the given key. */
+    override function put(K key, V value)
+        if not has(key)
+            keys.add(key)
+
+        super.put(key, value)
+
+
+    /** Get the number of mappings in this LinkedHashMap. */
+    function size() returns int
+        return keys.getSize()
+
+
+    /** Removes the value saved under the given key. */
+    override function remove(K key)
+        super.remove(key)
+        keys. remove(key)
+
+
+    /** Clears the LinkedHashMap of all entries */
+    override function flush()
+        super.flush()
+        if not keys == null
+            destroy keys
+            keys = new LinkedList<K>()
+
+
+    /** Gets an iterator for this LinkedHashMap. */
+    function iterator() returns LHMIterator<K, V>
+        return new LHMIterator(this, keys)
+
+
+    ondestroy
+        destroy keys
+        destroy dummy
+
+
+public class LHMEntry<K, V>
+    K key
+    V value
+
+    construct(K key, V value)
+        this.key   = key
+        this.value = value
+
+
+public class LHMIterator<K, V>
+    LHMEntry<K, V>      dummy
+    LinkedList<K>       keys
+    LLIterator<K>       keysIterator
+    LinkedHashMap<K, V> parentMap
+
+    construct(LinkedHashMap<K, V> parentMap, LinkedList<K> keys)
+        this.dummy        = new LHMEntry<K, V>(null, null)
+        this.keys         = keys
+        this.keysIterator = keys.iterator()
+        this.parentMap    = parentMap
+
+
+    /** Remove the current element from the LinkedHashMap. */
+    function remove()
+        if not dummy.key == null
+            parentMap.remove(dummy.key)
+            keysIterator.remove()
+
+
+    function hasNext() returns bool
+        return keysIterator.hasNext()
+
+
+    function next() returns LHMEntry<K, V>
+        dummy.key   = keysIterator.next()
+        dummy.value = parentMap.get(dummy.key)
+        return dummy
+
+
+    function close()
+        keysIterator.close()
+        destroy this
```

Comment:
The dummy should be destroyed at the end. It would be even nicer to use a tuple instead of the dummy element, but that would require generic tuples in Wurst :/